### ISSUE TITLE
Patch 12.17 Garena In-Game lookups bugged

### ISSUE THREAD

**Issue by downthecrop** [LyticalApp/Lytical#8]
![Screenshot_20220915-012445_Discord](https://user-images.githubusercontent.com/4764745/190210136-41517dfd-3ed4-457d-9baa-4ec7479622ab.png)

**Comment by downthecrop** [comment]
Haven't started on this so anyone is welcome to resolve it but I'll get around to it sometime soon-ish.

**Comment by downthecrop** [comment]
`/lol-gameflow/v1/session` has had its teamOne and teamTwo attributes emptied. These two lists used to show information such as summoner ID and selected champion. Now they don't show anything. I have searched for other endpoints which would provide the information but there don't seem to be any. You are also unable to get stats from `/lol-match-history/v1/products/lol/` before the match has ended and finalized. This seems like a deliberate removal.

The next step is to check if the data is available from the Client API instead of the LCU. I don't yet have any tooling to test these endpoints but I would assume that this is something that can be easily accessed.

```
{
  "gameClient": {
    "observerServerIp": "spectator-consumer.jp1.lol.pvp.net",
    "observerServerPort": 80,
    "running": true,
    "serverIp": "43.229.66.106",
    "serverPort": 5141,
    "visible": false
  },
  "gameData": {
    "gameId": 370220937,
    "gameName": "",
    "isCustomGame": false,
    "password": "",
    "playerChampionSelections": [],
    "queue": {
      "allowablePremadeSizes": [
        1,
        2,
        3,
        4,
        5
      ],
      "areFreeChampionsAllowed": true,
      "assetMutator": "",
      "category": "PvP",
      "championsRequiredToPlay": 16,
      "description": "ARAM",
      "detailedDescription": "",
      "gameMode": "ARAM",
      "gameTypeConfig": {
        "advancedLearningQuests": false,
        "allowTrades": true,
        "banMode": "SkipBanStrategy",
        "banTimerDuration": 0,
        "battleBoost": true,
        "crossTeamChampionPool": false,
        "deathMatch": false,
        "doNotRemove": false,
        "duplicatePick": false,
        "exclusivePick": true,
        "id": 21,
        "learningQuests": false,
        "mainPickTimerDuration": 0,
        "maxAllowableBans": 0,
        "name": "GAME_CFG_TEAM_BUILDER_RANDOM",
        "onboardCoopBeginner": false,
        "pickMode": "AllRandomPickStrategy",
        "postPickTimerDuration": 33,
        "reroll": false,
        "teamChampionPool": false
      },
      "id": 450,
      "isRanked": false,
      "isTeamBuilderManaged": true,
      "isTeamOnly": false,
      "lastToggledOffTime": 1633501920283,
      "lastToggledOnTime": 1633517422776,
      "mapId": 12,
      "maxLevel": 0,
      "maxSummonerLevelForFirstWinOfTheDay": 0,
      "maximumParticipantListSize": 5,
      "minLevel": 6,
      "minimumParticipantListSize": 1,
      "name": "ARAM",
      "numPlayersPerTeam": 5,
      "queueAvailability": "Available",
      "queueRewards": {
        "isChampionPointsEnabled": true,
        "isIpEnabled": true,
        "isXpEnabled": true,
        "partySizeIpRewards": []
      },
      "removalFromGameAllowed": false,
      "removalFromGameDelayMinutes": 0,
      "shortName": "ARAM",
      "showPositionSelector": false,
      "spectatorEnabled": true,
      "type": "ARAM_UNRANKED_5x5"
    },
    "spectatorsAllowed": false,
    "teamOne": [],
    "teamTwo": []
  },
  "gameDodge": {
    "dodgeIds": [],
    "phase": "None",
    "state": "Invalid"
  },
  "map": {
    "assets": {
      "champ-select-background-sound": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/sound/music-cs-allrandom-howlingabyss.ogg",
      "champ-select-banphase-background-sound": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/sound/music-cs-allrandom-banphase-howlingabyss.ogg",
      "champ-select-flyout-background": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/champ-select-flyout-background.png",
      "game-select-icon-active": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/game-select-icon-active.png",
      "game-select-icon-active-video": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/video/game-select-icon-active.webm",
      "game-select-icon-default": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/game-select-icon-default.png",
      "game-select-icon-disabled": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/game-select-icon-disabled.png",
      "game-select-icon-hover": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/game-select-icon-hover.png",
      "game-select-icon-intro-video": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/video/game-select-icon-intro.webm",
      "gameflow-background": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/gameflow-background.jpg",
      "gameflow-background-dark": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/gameflow-background-dark.jpg",
      "gameselect-button-hover-sound": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/Shared/sound/sfx-gameselect-button-hover.ogg",
      "icon-defeat": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-defeat.png",
      "icon-defeat-v2": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-defeat-v2.png",
      "icon-defeat-video": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/video/icon-defeat.webm",
      "icon-empty": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-empty.png",
      "icon-hover": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-hover.png",
      "icon-leaver": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-leaver.png",
      "icon-leaver-v2": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-leaver-v2.png",
      "icon-loss-forgiven-v2": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-loss-forgiven-v2.png",
      "icon-v2": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-v2.png",
      "icon-victory": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/icon-victory.png",
      "icon-victory-video": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/video/icon-victory.webm",
      "music-inqueue-loop-sound": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/sound/music-inqueue-loop-howlingabyss.ogg",
      "parties-background": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/parties-background.jpg",
      "postgame-ambience-loop-sound": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/sound/sfx-ambience-loop-howlingabyss.ogg",
      "ready-check-background": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/ready-check-background.png",
      "ready-check-background-sound": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/sound/sfx-readycheck-ha-portal.ogg",
      "sfx-ambience-pregame-loop-sound": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/sound/sfx-ambience-loop-howlingabyss.ogg",
      "social-icon-leaver": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/social-icon-leaver.png",
      "social-icon-victory": "lol-game-data/assets/content/src/LeagueClient/GameModeAssets/ARAM/img/social-icon-victory.png"
    },
    "categorizedContentBundles": {},
    "description": "The Howling Abyss is a bottomless crevasse located in the coldest, cruelest, part of the Freljord. Legends say that, long ago, a great battle took place here on the narrow bridge spanning this chasm. No one remembers who fought here, or why, but it is said that if you listen carefully to the wind you can still hear the cries of the vanquished tossed howling into the Abyss.",
    "gameMode": "ARAM",
    "gameModeName": "ARAM",
    "gameModeShortName": "ARAM",
    "gameMutator": "",
    "id": 12,
    "isRGM": false,
    "mapStringId": "HA",
    "name": "Howling Abyss",
    "perPositionDisallowedSummonerSpells": {},
    "perPositionRequiredSummonerSpells": {},
    "platformId": "",
    "platformName": "",
    "properties": {
      "suppressRunesMasteriesPerks": false
    }
  },
  "phase": "InProgress"
}
```

**Comment by downthecrop** [comment]
Looks like `GET ​https://127.0.0.1:2999/liveclientdata/playerlist` is what we're looking for.

**Comment by downthecrop** [comment]
```
[
	{
		"championName": "Draven",
		"isBot": false,
		"isDead": false,
		"items": [
			{
				"canUse": true,
				"consumable": false,
				"count": 1,
				"displayName": "Stealth Ward",
				"itemID": 3340,
				"price": 0,
				"rawDescription": "GeneratedTip_Item_3340_Description",
				"rawDisplayName": "Item_3340_Name",
				"slot": 6
			}
		],
		"level": 4,
		"position": "",
		"rawChampionName": "game_character_displayname_Draven",
		"rawSkinName": "game_character_skin_displayname_Draven_6",
		"respawnTimer": 0.0,
		"runes": {
			"keystone": {
				"displayName": "Lethal Tempo",
				"id": 8008,
				"rawDescription": "perk_tooltip_LethalTempo",
				"rawDisplayName": "perk_displayname_LethalTempo"
			},
			"primaryRuneTree": {
				"displayName": "Precision",
				"id": 8000,
				"rawDescription": "perkstyle_tooltip_7201",
				"rawDisplayName": "perkstyle_displayname_7201"
			},
			"secondaryRuneTree": {
				"displayName": "Domination",
				"id": 8100,
				"rawDescription": "perkstyle_tooltip_7200",
				"rawDisplayName": "perkstyle_displayname_7200"
			}
		},
		"scores": {
			"assists": 0,
			"creepScore": 0,
			"deaths": 0,
			"kills": 0,
			"wardScore": 0.0
		},
		"skinID": 6,
		"skinName": "Draven Draven",
		"summonerName": "ttv downthecrop",
		"summonerSpells": {
			"summonerSpellOne": {
				"displayName": "Ignite",
				"rawDescription": "GeneratedTip_SummonerSpell_SummonerDot_Description",
				"rawDisplayName": "GeneratedTip_SummonerSpell_SummonerDot_DisplayName"
			},
			"summonerSpellTwo": {
				"displayName": "Flash",
				"rawDescription": "GeneratedTip_SummonerSpell_SummonerFlash_Description",
				"rawDisplayName": "GeneratedTip_SummonerSpell_SummonerFlash_DisplayName"
			}
		},
		"team": "ORDER"
	}
]
```

**Comment by downthecrop** [comment]
resolved by #9
